internal/semaphore/semaphore_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package semaphore
+
+import (
+	"sync"
+	"testing"
+	"github.com/stretchr/testify/assert"
+)
+
+func checkLimit(t *testing.T, sem Semaphore, expected int) {
+	limit := sem.GetLimit()
+	assert.Equal(t, expected, limit)
+}
+
+func checkCount(t *testing.T, sem Semaphore, expected int) {
+	count := sem.GetCount()
+	assert.Equal(t, expected, count)
+}
```

Comment:
Neither of these functions get used outside of the `checkLimitAndCount` function. They are not useful to separate.